src/stores/GuildMemberRoleStore.js

```diff
@@ -138,6 +149,24 @@ class GuildMemberRoleStore extends Collection {
    *   .catch(console.error);
    */
   set(roles, reason) {
+    if (roles === undefined) return Promise.resolve(this.member);
+    if (!(roles instanceof Collection || Array.isArray(roles))) {
```

Comment:
![image](https://user-images.githubusercontent.com/17960496/60201321-a71cd100-9850-11e9-8355-b1abfacf695a.png)


```suggestion
    if (!((roles instanceof Collection) || Array.isArray(roles))) {
```

Reply:
Changed, also moved the is array portion inside the other if statement and had to move the map outside the collection constructor to return a promise rejection.